explore/src/main/scala/explore/proposal/ProposalTabContents.scala

```diff
@@ -79,113 +76,120 @@ object ProposalTabContents:
-          .zoom(ProposalInfo.optProposal)
-          .mapValue((proposalView: View[Proposal]) =>
-            ProposalEditor(
-              programId,
-              proposalView,
-              undoStacks,
-              minExecutionTime,
-              maxExecutionTime,
-              users,
-              attachments,
-              userVault.map(_.token)
-            ): VdomNode
-          )
-          .getOrElse(userVault.map(_.user) match {
-            case Some(_: StandardUser) =>
-              <.div(
-                ExploreStyles.HVCenter,
-                Button(
-                  label = "Create a Proposal",
-                  icon = Icons.FileCirclePlus.withClass(LoginStyles.LoginOrcidIcon),
-                  clazz = LoginStyles.LoginBoxButton,
-                  severity = Button.Severity.Secondary,
-                  onClick = createProposal(
-                    programId,
-                    optProposalInfo,
-                    minExecutionTime,
-                    maxExecutionTime,
-                    timeCharge
-                  )
-                ).big
+    val isStdUser      = userVault.map(_.user).collect { case _: StandardUser => () }.isDefined
+    val proposalStatus = programDetails.get.proposalStatus
+
+    def updateStatus(newStatus: ProposalStatus): Callback =
+      (for {
+        _ <- ProgramQueries.updateProposalStatus[IO](programId, newStatus)
+        _ <- programDetails.zoom(ProgramDetails.proposalStatus).set(newStatus).toAsync
+      } yield ()).switching(isUpdatingStatus.async, IsUpdatingStatus(_)).runAsync
+
+    programDetails
+      .zoom(ProgramDetails.proposal)
+      .mapValue((proposalView: View[Proposal]) =>
+        <.div(
+          ExploreStyles.ProposalTab,
+          ProposalEditor(
+            programId,
+            userVault.map(_.user.id),
+            proposalView,
+            undoStacks,
+            minExecutionTime,
+            maxExecutionTime,
+            users,
+            attachments,
+            userVault.map(_.token),
+            layout
+          ),
+          Toolbar(left =
+            <.span(
+              Tag(
+                value = programDetails.get.proposalStatus.name,
+                severity =
+                  if (proposalStatus === ProposalStatus.Accepted) Tag.Severity.Success
+                  else Tag.Severity.Danger
               )
-            case _                     =>
-              <.div(
-                ExploreStyles.HVCenter,
-                Button(
-                  label = "Login with ORCID to create a Proposal",
-                  icon = Image(src = Resources.OrcidLogo, clazz = LoginStyles.LoginOrcidIcon),
-                  clazz = LoginStyles.LoginBoxButton,
-                  severity = Button.Severity.Secondary,
-                  onClick = ctx.sso.switchToORCID.runAsync
-                ).big
+                .when(proposalStatus > ProposalStatus.Submitted),
+              // TODO: Validate proposal before allowing submission
```

Comment:
Will this be done on the client side?

Reply:
I think it should be done on both the client and server side.